tokens/usecase/tokens_usecase_test.go

```diff
@@ -135,6 +142,21 @@ func (s *TokensUseCaseTestSuite) TestParseExponents_Testnet() {
 	s.Require().Equal(defaultCosmosExponent, osmoToken.Precision)
 }
 
+func (s *TokensUseCaseTestSuite) TestGetPrices_Coingecko() {
+	// Set up mainnet mock state.
+	mainnetUsecase := s.SetupDefaultRouterAndPoolsUsecase()
+	prices, err := mainnetUsecase.Tokens.GetPrices(context.Background(), routertesting.MainnetDenoms, []string{USDC}, domain.CoinGeckoPricingSourceType)
```

Comment:
I would like to note that this test is making network requests to Coingecko. Instead, we should monkey-patch a mock function that would simulate the interaction with Coingecko.

It would be challenging to relay in words what I mean so I did a quick demo here:
https://github.com/osmosis-labs/sqs/pull/218

Please let me know if you have any questions - we can discuss during data sync tomorrow